lib/datadog/core/telemetry/http/transport.rb

```diff
@@ -13,18 +14,47 @@ module Http
         # Class to send telemetry data to Telemetry API
         # Currently only supports the HTTP protocol.
         class Transport
+          def self.build_agent_transport
+            agent_settings = Configuration::AgentSettingsResolver.call(Datadog.configuration)
+
+            Transport.new(
+              host: agent_settings.hostname,
+              port: agent_settings.port,
+              path: Http::Ext::AGENT_ENDPOINT
+            )
+          end
+
+          def self.build_agentless_transport(api_key:, dd_site:)
+            host =
+              if dd_site == Environment::Ext::DD_SITE_STAGING
+                # special case for staging - the url is constructed differently
+                'all-http-intake.logs.datad0g.com'
+              else
```

Comment:
Hmm... I'm not the hugest fan of having internal-only-exceptions in the production code for the library. Since this is only for testing, may I suggest adding a kind of an override somewhere where you can specify the full host, so we can use it here?

That way this code wouldn't need to know about staging stuff, just that optionally you can override the entire url. (I think we could even add such an override directly to the `AgentSettingsResolver` even? -- add arguments for agentless telemetry config)